Description: Printing should not error
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: not-needed
Last-Update: 2024-07-05

Index: libjxl/tools/ssimulacra2_main.cc
===================================================================
--- libjxl.orig/tools/ssimulacra2_main.cc
+++ libjxl/tools/ssimulacra2_main.cc
@@ -48,7 +48,7 @@ int PrintUsage(char** argv) {
   fprintf(stderr,
           "                             average output of cjxl -q 90 or "
           "mozjpeg -quality 90)\n");
-  return 1;
+  return 0;
 }
 
 int main(int argc, char** argv) {
